src/components/organisms/UiList/UiList.vue

```diff
@@ -45,53 +45,39 @@
 
 <script setup lang="ts">
 import { computed } from 'vue';
-import type { PropType } from 'vue';
-import type { Icon } from '../../../types/icon';
-import type { ListTag } from '../../../types/tag';
+import type { HTMLAttributes } from 'vue';
 import UiListItem from './_internal/UiListItem.vue';
+import type { ListItemAttrsProps } from './_internal/UiListItem.vue';
 import UiText from '../../atoms/UiText/UiText.vue';
+import type {
+  DefineAttrsProps,
+  ListHTMLTag,
+} from '../../../types';
 
-export interface ListItemSuffixAttrs {
-  label?: string;
-  icon?: Icon;
-  iconSuffixAttrs?: Record<string, unknown>;
-  [key: string]: unknown;
-}
-export interface ListItemAttrs {
-  icon?: Icon;
-  hasSuffix?: boolean;
-  suffixAttrs?: ListItemSuffixAttrs;
-  [key: string]: unknown;
-}
-export interface ListItemComplex {
+export interface ListRenderItem extends ListItemAttrsProps {
   name: string;
   label?: string;
-  children?: {
-    tag?: ListTag;
-    items?: (string | ListItemComplex)[];
-    listItemAttrs?: ListItemAttrs;
-  }
-  listItemAttrs?: ListItemAttrs;
+  // eslint-disable-next-line no-use-before-define
```

Comment:
Q: Why this comment and not changing the order of defining? 🤔 😁

Reply:
IMO Changing the order won't make a difference because the rest of the types are also interdependent.